internal/pkg/server/fleet_integration_test.go

```diff
@@ -379,6 +381,7 @@ func TestServerConfigErrorReload(t *testing.T) {
 		cancel()
 	}).Return(nil)
 	mReporter.On("UpdateState", client.UnitStateStopping, mock.Anything, mock.Anything).Return(nil)
+	mReporter.On("UpdateState", client.UnitStateFailed, mock.MatchedBy(func(err error) bool { return errors.Is(err, context.Canceled) }), mock.Anything).Return(nil).Maybe()
```

Comment:
I found that without this in the mocked calls list the test will panic because the `cancel()` call when the state updates to healthy interrupts something and returns the context cancelled error triggering a fail-state detection instead of a stopping call